### PR TITLE
Feature/sg 109 beit architecture

### PR DESCRIPTION
**Pull request by shaydeci**

### CODE REVIEW

**Review by ofrimasad — COMMENTED** [review]
a few comments inside

File: tests/integration_tests/pretrained_models_test.py [hunk 1]
```diff
@@ -48,15 +48,16 @@ def setUp(self) -> None:
...
+                                               "beit_base_patch16_224": 0.85
                                                }
         self.imagenet_dataset = ImageNetDatasetInterface(data_dir="/data/Imagenet", dataset_params={"batch_size": 128})
 
         self.imagenet_dataset_05_mean_std = ImageNetDatasetInterface(data_dir="/data/Imagenet",
                                                                      dataset_params={"batch_size": 128,
                                                                                      "img_mean": [0.5, 0.5, 0.5],
                                                                                      "img_std": [0.5, 0.5, 0.5],
-                                                                                     "resize_size": 249
+                                                                                     "resize_size": 248
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply tests/integration_tests/pretrained_models_test.py:60]: Vits are indeed trained with a weird crop ratio....

[the remark this answers is not in the document]
Comment (shaydeci) [reply tests/integration_tests/pretrained_models_test.py:60]: Vits are trained with a weird crop ratio indeed...

File: src/super_gradients/training/models/classification_models/beit.py [hunk 2]
```diff
@@ -0,0 +1,465 @@
...
+        gs_new = [int(math.sqrt(ntok_new))] * 2
+    assert len(gs_new) >= 2
+    posemb_grid = posemb_grid.reshape(1, gs_old, gs_old, -1).permute(0, 3, 1, 2)
+    posemb_grid = F.interpolate(posemb_grid, size=gs_new, mode='bicubic', align_corners=False)
+    posemb_grid = posemb_grid.permute(0, 2, 3, 1).reshape(1, gs_new[0] * gs_new[1], -1)
+    posemb = torch.cat([posemb_tok, posemb_grid], dim=1)
+    return posemb
+
+
+def checkpoint_filter_fn(state_dict, model):
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/training/models/classification_models/beit.py]: removed

File: src/super_gradients/training/models/classification_models/beit.py [hunk 3]
```diff
@@ -0,0 +1,465 @@
...
+        elif k == 'pos_embed' and v.shape != model.pos_embed.shape:
+            # To resize pos embedding when using model at different size from pretrained weights
+            v = resize_pos_embed(
+                v, model.pos_embed, getattr(model, 'num_tokens', 1), model.patch_embed.grid_size)
+        out_dict[k] = v
+    return out_dict
+
+
+def _no_grad_trunc_normal_(tensor, mean, std, a, b):
+    # Cut & paste from PyTorch official master until it's in a few official releases - RW
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/training/models/classification_models/beit.py:59]: done

File: src/super_gradients/training/models/all_architectures.py [hunk 4]
```diff
@@ -115,6 +115,8 @@
                  "vit_base": vit_base,
                  "vit_large": vit_large,
                  "vit_huge": vit_huge,
+                 "beit_base_patch16_224": beit_base_patch16_224,
+                 "beit_large_patch16_224": beit_large_patch16_224
```
Comment (shairoz-deci) [inline src/super_gradients/training/models/all_architectures.py:119]: we should have some consistency with ViTs. if we say "vit_base" and not "vit_base_patchX_X" then beit should be similar or the ViTs should be changed

Comment (shaydeci) [reply src/super_gradients/training/models/all_architectures.py:119]: Agree, Ill change the Vitt's strings for now.

**Review by ofrimasad — APPROVED** [review]
LGTM

**Pull request merged**